publication-rest/src/main/java/no/unit/nva/publication/download/ImportCandidatePresignedUrlHandler.java

```diff
@@ -0,0 +1,87 @@
+import java.net.HttpURLConnection;
+import java.util.UUID;
+import no.unit.nva.identifiers.SortableIdentifier;
+import no.unit.nva.model.associatedartifacts.file.File;
+import no.unit.nva.publication.RequestUtil;
+import no.unit.nva.publication.model.business.importcandidate.ImportCandidate;
+import no.unit.nva.publication.service.impl.ResourceService;
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import nva.commons.apigateway.exceptions.NotFoundException;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.presigner.S3Presigner;
+
+public class ImportCandidatePresignedUrlHandler extends ApiGatewayHandler<Void, PresignedUrl> {
+
+    public static final String BUCKET = new Environment().readEnv("IMPORT_CANDIDATES_STORAGE_BUCKET");
+    public static final String IMPORT_CANDIDATE_MISSES_FILE_EXCEPTION_MESSAGE = "Import candidate with identifier: %s"
+                                                                                + " \nDoes not have file with "
+                                                                                + "identifier: %s";
+    private final S3Presigner s3Presigner;
+    private final ResourceService importCandidateService;
+
+    @JacocoGenerated
+    public ImportCandidatePresignedUrlHandler() {
+        super(Void.class);
+        this.importCandidateService = ResourceService.defaultService();
+        this.s3Presigner = defaultS3Presigner();
+    }
+
+    public ImportCandidatePresignedUrlHandler(S3Presigner s3Presigner, ResourceService importCandidateService) {
+        super(Void.class);
+        this.s3Presigner = s3Presigner;
+        this.importCandidateService = importCandidateService;
+    }
+
+    @Override
+    protected PresignedUrl processInput(Void unused, RequestInfo requestInfo, Context context)
+        throws ApiGatewayException {
+        var importCandidateIdentifier = RequestUtil.getImportCandidateIdentifier(requestInfo);
+        var fileIdentifier = RequestUtil.getFileIdentifier(requestInfo);
+
+        validateExistence(importCandidateIdentifier, fileIdentifier);
+
+        return PresignedUrl.fromS3Key(fileIdentifier).bucket(BUCKET).create(s3Presigner);
+    }
+
+    @Override
+    protected Integer getSuccessStatusCode(Void unused, PresignedUrl o) {
+        return HttpURLConnection.HTTP_OK;
+    }
+
+    private void validateExistence(SortableIdentifier importCandidateIdentifier, UUID fileIdentifier)
+        throws NotFoundException {
+        var importCandidate = importCandidateService.getImportCandidateByIdentifier(importCandidateIdentifier);
+        if (fileExists(importCandidate, fileIdentifier)) {
+            return;
+        }
```

Comment:
if you negate this and put the exception in the condition body, it would be cleaner, I guess